src/components/CategoryDropdown/CategoryDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from "react";
+import { Link } from "react-router-dom";
+import "@/src/styles/index.scss";
+
+interface CategoryDropdownProps {
+  onSelectCategory: (category: string) => void;
+}
+
+const CategoryDropdown: React.FC<CategoryDropdownProps> = ({
+  onSelectCategory,
+}) => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const toggleMenu = () => {
+    setIsMenuOpen(!isMenuOpen);
+  };
+
+  const closeMenu = () => {
+    setIsMenuOpen(false);
+  };
+  const handleCategoryClick = (category: string) => {
+    onSelectCategory(category);
+    closeMenu();
+  };
+  return (
+    <div className="relative">
+      <button onClick={toggleMenu} className="flex items-center ">
+        <h2 className="text-xl font-bold mb-2">Category</h2>
+        <svg
+          className="h-6 w-6 inline-block"
+          fill="none"
+          strokeLinecap="round"
+          strokeLinejoin="round"
+          strokeWidth="2"
+          viewBox="0 0 24 24"
+          stroke="currentColor"
+        >
+          {isMenuOpen ? (
```

Comment:
Usually you try to keep the rendering logic out of the render. One liners are sometimes ok depending on the team.